src/endpoint/Timesheet.Delete/FlowStep/Step.DeleteTimesheet.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Concurrent;
+using GarageGroup.Infra.Bot.Builder;
+using Microsoft.Extensions.Logging;
+
+namespace GarageGroup.Internal.Timesheet;
+
+partial class DateTimesheetFlowStep
+{
+    internal static ChatFlow<DeleteTimesheetFlowState> DeleteTimesheet(
+        this ChatFlow<DeleteTimesheetFlowState> chatFlow, ICrmTimesheetApi timesheetApi)
+        =>
+        chatFlow.ForwardValue(timesheetApi.Delete);
+
+    private static ValueTask<ChatFlowJump<DeleteTimesheetFlowState>> Delete(
+        this ICrmTimesheetApi crmTimesheetApi, IChatFlowContext<DeleteTimesheetFlowState> context, CancellationToken cancellationToken)
+        =>
+        AsyncPipeline.Pipe(
+            context.FlowState, cancellationToken)
+        .Pipe(
+            state => (state, crmTimesheetApi))
+        .PipeValue(
+            DeleteTimesheetAsync)
+        .OnFailure(
+            context.Logger.LogDeleteFailures)
+        .Map(
+            _ => context.FlowState,
+            ToBreakState)
+        .Fold(
+            ChatFlowJump.Next,
+            ChatFlowJump.Break<DeleteTimesheetFlowState>);
+
+    private static void LogDeleteFailures (this ILogger logger, DeleteFailure deleteFailure)
+    {
+        foreach (var failure in deleteFailure.Failures)
+        {
+            logger.LogError(failure.Failure.SourceException, 
+                "Не удалось удалить timesheet. TimesheetId: {0}. FailureMessage: {1}. FailureCode: {2}"
+                , failure.Id, failure.Failure.FailureMessage, failure.Failure.FailureCode);
+        }
+    }
+
+    private static async ValueTask<Result<Unit, DeleteFailure>> DeleteTimesheetAsync(
+        (DeleteTimesheetFlowState State, ICrmTimesheetApi CrmTimesheetApi) input, CancellationToken cancellationToken)
+    {
+        var failures = new ConcurrentBag<(Failure<TimesheetDeleteFailureCode> Failure, Guid Id)>();
+        await Parallel.ForEachAsync(input.State.DeleteTimesheetsId.AsEnumerable(), InnerDeleteAsync);
```

Comment:
Use `ForEachAsync` overloading that accept options and pass cancellation token into options object